python/ccxt/async_support/bitrue.py

```diff
@@ -0,0 +1,382 @@
+        return self.parse_trades(data, market, since, limit)
+
+    def parse_trade(self, trade, market=None):
+        side = 'buy' if self.safe_value(trade, 'isBuyer') else 'sell'
+        takerOrMaker = 'maker' if self.safe_value(trade, 'isMaker') else 'taker'
+        symbol = None
+        if market is not None:
+            symbol = market['symbol']
+        if symbol is None:
+            if market is None:
+                market = self.markets_by_id[self.safe_string_lower(trade, 'symbol')]
+            symbol = market['symbol']
+        timestamp = self.safe_integer(trade, 'time')
+        if timestamp is None:
+            timestamp = self.parse8601(self.safe_string(trade, 'time'))
+        price = self.safe_float(trade, 'price')
+        amount = self.safe_float(trade, 'qty')
+        cost = price * amount
+        return {
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': self.iso8601(timestamp),
+            'symbol': symbol,
+            'id': self.safe_string(trade, 'id'),
+            'order': None,
+            'type': 'limit',
+            'takerOrMaker': takerOrMaker,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': cost,
+            'fee': None,
+        }
+
+    async def fetch_time(self, params={}):
+        response = await self.publicGetTime(params)
+        return self.safe_integer(response, 'serverTime')
+
+    async def fetch_balance(self, params={}):
+        await self.load_markets()
+        response = await self.privateGetAccount(params)
+        balances = self.safe_value(response, 'balances')
+        result = {'info': response}
+        for i in range(0, len(balances)):
+            balance = balances[i]
+            currencyId = self.safe_value(balance, 'asset')
+            code = self.safe_currency_code(currencyId)
+            account = self.account()
+            account['free'] = self.safe_float(balance, 'free')
+            account['used'] = self.safe_float(balance, 'locked')
+            result[code] = account
+        return self.parse_balance(result)
+
+    async def load_time_difference(self, params={}):
+        serverTime = await self.fetch_time(params)
+        after = self.milliseconds()
+        self.options['timeDifference'] = after - serverTime
+        return self.options['timeDifference']
+
+    async def fetch_closed_orders(self, symbol=None, start_id=None, limit=None, params={}):
+        await self.load_markets()
+        if symbol is not None:
+            market = self.market(symbol)
+        request = {}
+        if symbol is not None:
+            request['symbol'] = market['id']
+        if start_id is not None:
+            request['fromId'] = start_id
+        if limit is not None:
+            request['limit'] = limit
+        response = self.privateGetMyTrades(self.extend(request, params))
+        trades = response if isinstance(response, (list)) else []
```

Comment:
Should always be a list right?
```suggestion
        trades = response if isinstance(response, list) else []
```

Reply:
Hi @mrbaguvix , I believe this was generated. I also didn't commit this file (I mentioned in the PR description that the compiled assets were added by previous contributors, whose work I started from), so perhaps @firepol can weigh in here. The python files will have to be rebuilt when the PR is ready to be merged (or post-merge, I'm not sure how that works).

The contribution guidelines state:

```
There is now a single source version in one language, that is JavaScript. Other language-specific versions are syntactically derived (transpiled, generated) automatically from the source version. 
```

So I took this to mean that I shouldn't commit any changes from rebuilding.

Reply:
in fact, the section I linked in the description instructs us *not* to commit python and other transpiled assets.